configure.ac

```diff
@@ -267,29 +254,102 @@ dnl --- enable_pkey
 enable_pkey=no
 
 case $target in
-     *s390x*)
-	  enable_pkey=yes
-	  ;;
-     *s390*)
-	  CFLAGS="$CFLAGS -m31 -DNO_PKEY"
-	  ;;
-     *ppc64* | *x86_64*)
-	  CFLAGS="$CFLAGS -DNO_PKEY"
-	  ;;
-     *ppc* | i*86*)
-	  CFLAGS="$CFLAGS -m32 -DNO_PKEY"
-	  ;;
-     *)
-	  CFLAGS="$CFLAGS -DNO_PKEY"
-	  ;;
+	*s390x*)
+		enable_pkey=yes
+	;;
+	*s390*)
+		CFLAGS="$CFLAGS -m31"
+		AC_DEFINE([NO_PKEY])
+	;;
+	*ppc64* | *x86_64*)
+		AC_DEFINE([NO_PKEY])
+	;;
+	*ppc* | i*86*)
+	  CFLAGS="$CFLAGS -m32"
+		AC_DEFINE([NO_PKEY])
+	;;
+	*)
+		AC_DEFINE([NO_PKEY])
+	;;
 esac
 AM_CONDITIONAL([NO_PKEY], [test "x$enable_pkey" != "xyes"])
 
+dnl Check and save whether the platform is AIX
+dnl Using the pthread library requires the _THREAD_SAFE macro to be set
+build_aix=no
+case "${host_os}" in
+	aix*)
+		build_aix=yes
+		AC_DEFINE([_ALL_SOURCE])
+		AC_DEFINE([_THREAD_SAFE])
+		with_systemd=no
+		with_libica=no
+		enable_ccatok=yes
+		enable_swtok=yes
+		enable_tpmtok=no
+		enable_icsftok=no
+		enable_ep11tok=no
+		echo "Disabling migration utilities on AIX: pkcscca pkcstok_migrate"
+		enable_pkcscca=no
+		enable_pkcstok_migrate=no
+		echo "Disabling the following unsupported features on AIX: systemd libica tpmtok icsftok ep11tok"
+		pkcsslotd_user=slotd
+		echo "Shortening username to $pkcsslotd_user due to platform restrictions"
+		function parse_map {
+			grep '^        ' $1 | tr -d ';' | tr -d ' ' > "$1".sym
+		}
+		parse_map opencryptoki.map
+		parse_map opencryptoki_tok.map
+		;;
+esac
+
+dnl Define custom variables
+lockdir=${localstatedir}/lock/opencryptoki
+logdir=${localstatedir}/log/opencryptoki
+
+if test "x$build_aix" = "xyes"; then
+	run_path=${runstatedir}
```

Comment:
Isn't this the opposite of what we want? 
On Linux it was and should be `/run` (i..e `${runstatedir}`), but on AIX it should be `/var/run`, no? 
The way you have it now is that its `/var/run` on Linux (we don't want that!), but `${runstatedir}` on AIX....

Reply:
`/var/run` is a symlink to /run on systems that have adopted the new `rundir` format. 
Are there systems where that is not the case?

Again, on RHEL 8.9 (because that is the only Linux system I have at the moment):
```
(0) root @ rtc02: /root
# ls -l /var/run
lrwxrwxrwx. 1 root root 6 May  5  2023 /var/run -> ../run

(0) root @ rtc02: /root
# uname -r
4.18.0-513.18.1.el8_9.ppc64le
```

As for
>On Linux it was and should be /run (i..e ${runstatedir}), but on AIX it should be /var/run, no?

Nope, other way around - because tmpfs is a thing on Linux, it makes sense to hardcode it to use the tmpfs filesystem for all the expendable runtime artifacts it leaves behind. AIX does not have tmpfs, so the folder you use doesn't matter (`/var` is also a jfs2 mount), which is why I chose to have it fully contained within the `${prefix}` directory.

The sources already assumed `/var/run` as the directory path, I'm just codifying it and passing it down through autoconf.

Reply:
P.S. the changes also account for packaging usage of `DESTDIR` - not that it matters, since the folder contents are only generated when the daemon starts up.

Reply:
Sorry, but no. Under Linux is should be `/run`, NOT `/var/run`. We did change this with the following commit quite some time ago: https://github.com/opencryptoki/opencryptoki/commit/08eaca02385e61759743a912f2f3bb7f262d78cd
`/var/run` is deprecated and should no longer be used on Linux. 
Yes, `/var/run` is a symlink to `/run` on current distros.

So please check you code change once again. With your change it is no back to `/var/run` on Linux.